eZ/Publish/Core/FieldType/BinaryBase/Type.php

```diff
@@ -347,20 +347,7 @@ public function validateValidatorConfiguration( $validatorConfiguration )
             switch ( $validatorIdentifier )
             {
                 case 'FileSizeValidator':
-                    if ( !isset( $parameters['maxFileSize'] ) )
-                    {
-                        $validationErrors[] = new ValidationError(
-                            "Validator %validator% expects parameter %parameter% to be set.",
-                            null,
-                            array(
-                                "validator" => $validatorIdentifier,
-                                "parameter" => 'maxFileSize',
-                            ),
-                            "[$validatorIdentifier][maxFileSize]"
-                        );
-                        break;
-                    }
-                    if ( !is_int( $parameters['maxFileSize'] ) && $parameters['maxFileSize'] !== false )
+                    if ( array_key_exists( "maxFileSize", $parameters ) && !is_int( $parameters['maxFileSize'] ) && $parameters['maxFileSize'] !== null )
```

Comment:
Actually, an error should still be raised if `maxFileSize` is absent, so you should keep the if block above, changing `isset()` to `array_key_exists()`. Then you don't have to remove the invalid provider part in `BinaryBaseTest::provideInvalidValidatorConfiguration()`